util.go

```diff
@@ -101,6 +101,73 @@ func BuildGarpPkt(ip net.IP, mac net.HardwareAddr, vlanID uint16) *openflow13.Pa
 	return pktOut
 }
 
+func ipChecksum(data []byte) uint16 {
+	csum := uint32(0)
+	length := len(data) - 1
+	for i := 0; i < length; i += 2 {
+		csum += uint32(data[i]) << 8
+		csum += uint32(data[i+1])
+	}
+	if len(data)%2 == 1 {
+		csum += uint32(data[length]) << 8
+	}
+	for csum > 0xffff {
+		csum = (csum >> 16) + (csum & 0xffff)
+	}
+	return ^uint16(csum)
+}
+
+func buildDnsForwardPkt(ethInPkt *protocol.Ethernet) *protocol.Ethernet {
+	ipPkt := ethInPkt.Data.(*protocol.IPv4)
+	ethPkt := protocol.NewEthernet()
+	ethPkt.HWSrc = ethInPkt.HWSrc
+	ethPkt.HWDst = ethInPkt.HWDst
+	ethPkt.VLANID.VID = 4093 // magic vlan
+	ethPkt.Ethertype = ethInPkt.Ethertype
+	ethPkt.Data = ipPkt
+	return ethPkt
+}
+
+func buildUDPRespPkt(inEth *protocol.Ethernet, uData []byte) (*protocol.Ethernet, error) {
+	inIp := inEth.Data.(*protocol.IPv4)
+	inUdp := inIp.Data.(*protocol.UDP)
+
+	outUdp := protocol.NewUDP()
+	outUdp.PortDst = inUdp.PortSrc
+	outUdp.PortSrc = inUdp.PortDst
+	outUdp.Length = uint16(8 + len(uData))
+	outUdp.Data = append(outUdp.Data, uData...)
+
+	outIp := protocol.NewIPv4()
+	outIp.Version = inIp.Version
+	outIp.IHL = 5
```

Comment:
the options seem to be copied from the input packet. We should include the size of that so that we don't miss out on any scenarios.